### DISCUSSION TITLE
Counting contributions of a blocked users repository

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22502] [New to GitHub]
<p>Hello. Sry if this is not the right place. I’m new here.</p>
<p>Here is my problem:</p>
<p>I am working with a soziopath and I don’t have a choice. The project is almost done though. I could not get him to let me create the repository so I commited over 300 times in the past month to “his” repository (while he commited 30 times with 10 loc commits per commit btw)</p>
<p>I really want to block this guy everywhere I can, when the project is done, but as I’ve read from the details about blocking another user is that the contributions are not counted anymore towards any repository owned by him if I block him.</p>
<p>Having all the contributions would give me a way better appearance when applying for a job, as the github page is all the people look at where I come from. It would also be great to have the repository attached to my git page in some way. 99 % of the code is written by myself anyways.</p>
<p>What can I do ?</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="Bl4ck-orig" data-post="1" data-topic="230130">
<div class="title">
<div class="quote-controls"></div>
 Bl4ck-orig:</div>
<blockquote>
<p>I am working with a soziopath and I don’t have a choice.</p>
</blockquote>
</aside>
<p>These are two rather strong statements: you are (1) diagnosing your collaborator, and (2) claiming you are working under coercion. The first point is most likely just your personal opinion (unless you end up naming the other party, in which case you would be damaging his reputation and thus violating the GitHub Code of Conduct).</p>
<p>As for the second point, which seems more likely a figure of speech than a matter of fact statement, there is little anyone here can do for you (unless you are <em>truly</em> being coerced, in which case it’s a police related matter). FOSS collaboration is either the result of a working contract or voluntary efforts, so when things between working partners don’t work there’s only a solution really — “divorce”, which in case of a working contract means quitting the job, and in case of voluntary collaboration turned bitter implies breaking up the collaborative effort.</p>
<aside class="quote no-group" data-username="Bl4ck-orig" data-post="1" data-topic="230130">
<div class="title">
<div class="quote-controls"></div>
 Bl4ck-orig:</div>
<blockquote>
<p>I really want to block this guy everywhere I can, when the project is done, but as I’ve read from the details about blocking another user is that the contributions are not counted anymore towards any repository owned by him if I block him.</p>
</blockquote>
</aside>
<p>Then don’t block him, just ignore him. You can’t eat the cake and have it. If you block him, this will affect your contributions history — and, in any case, since he’s the repository owner he could at any time delete the repository, or squash all its commits into a single one, both of which would affect your contributions stats anyway.</p>
<aside class="quote no-group" data-username="Bl4ck-orig" data-post="1" data-topic="230130">
<div class="title">
<div class="quote-controls"></div>
 Bl4ck-orig:</div>
<blockquote>
<p>Having all the contributions would give me a way better appearance when applying for a job, as the github page is all the people look at where I come from.</p>
</blockquote>
</aside>
<p>Maybe so, but I think that in the working environment the ability to work well with others is probably valued above technical skills, since hiring companies expect team work. So probably this post could damage more your job application than lost contributions to third party projects (hiring companies also sift through your posts, to get to known you better).</p>
<aside class="quote no-group" data-username="Bl4ck-orig" data-post="1" data-topic="230130">
<div class="title">
<div class="quote-controls"></div>
 Bl4ck-orig:</div>
<blockquote>
<p>I come from. It would also be great to have the repository attached to my git page in some way. 99 % of the code is written by myself anyways.</p>
</blockquote>
</aside>
<p>All I can think of is keeping your fork of this repository, at least you’re sure that regardless of its faith with it owner you’ll always have a copy that witnesses your contributions — your fork will still show all your commit in the repository stats and history, which should be good enough.</p>
<p>Think of web designers, which all they are left of their past works is often nothing more than a screenshot, since their clients are free to redesign their website every year if the wish to. Even for them, all that’s left is a mere copy of their original glorious work. When you contribute to someone else’s project, of work for a client, you have to accept this from the onset (their the project, in their hands its faith).</p>
<p>In any case, what really matters are your skills and gained experience, and nothing nor no-one can take them away from you.</p>

**Comment by jfullstackdev** [comment]
<p>yup, commits can even be faked,</p>
<p>I highly suggest focus on your profile, you can make public projects that can showcase your skills, so when  it’s asked by the employer, you can simply give your GitHub Profile and they will see all of that,</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="jdevstatic" data-post="3" data-topic="230130">
<div class="title">
<div class="quote-controls"></div>
 jdevstatic:</div>
<blockquote>
<p>yup, commits can even be faked,</p>
</blockquote>
</aside>
<p><a class="mention" href="/jdevstatic">@jdevstatic</a> has a good point here! In fact, if you want to build a strong curriculum in that sense you should always sign your commits via GPG/PGP, as a proof that you’re the original author. But that’s too late for past commits.</p>

**Comment by ghost** [comment]
<p>Thank you both for your answer. I will do that then.</p>
